rosbag2_compression/src/rosbag2_compression/zstd_decompressor.cpp

```diff
@@ -0,0 +1,184 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <cstdio>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "rcpputils/filesystem_helper.hpp"
+
+#include "rosbag2_compression/zstd_decompressor.hpp"
+
+#include "rosbag2_storage/filesystem_helper.hpp"
+
+#include "logging.hpp"
+
+namespace
```

Comment:
Maybe move these to a common source file. But that's for another PR.

Reply:
The helper functions for reading a file into a buffer and writing a file into a buffer cannot be pulled out since their implementations will change once streaming is supported.